sources/packages/web/src/components/students/modals/CancelApplication.vue

```diff
@@ -0,0 +1,88 @@
+<template>
+  <modal-dialog-base
+    title="Cancel application"
+    :dialog-type="DialogTypes.question"
+    :showDialog="showDialog"
+    :max-width="730"
+  >
+    <template #content>
+      Cancelling your application will stop your application from going forward.
+      After cancelling, you can still view your application but will not be able
+      to make further edits to it.
+      <strong>Are you sure you want to proceed?</strong>
+    </template>
+    <template #footer>
+      <footer-buttons
+        primaryLabel="Cancel application now"
+        secondaryLabel="No"
+        @primaryClick="cancelApplication"
+        @secondaryClick="dialogClosed"
+      />
+    </template>
+  </modal-dialog-base>
+</template>
+
+<script lang="ts">
+import ModalDialogBase from "@/components/generic/ModalDialogBase.vue";
+import { useModalDialog, useSnackBar } from "@/composables";
+import { ApplicationService } from "@/services/ApplicationService";
+import {
+  ApplicationStatus,
+  ApplicationStatusToBeUpdatedDto,
+  DialogTypes,
+} from "@/types";
+import { defineComponent } from "vue";
+
+export default defineComponent({
+  components: {
+    ModalDialogBase,
+  },
+  setup() {
+    const {
+      showDialog,
+      resolvePromise,
+      showModal: showModalInternal,
+    } = useModalDialog<boolean>();
+    const snackBar = useSnackBar();
+
+    const dialogClosed = () => {
+      resolvePromise(false);
+    };
+
+    let applicationId = 0;
+
+    // Show the modal and loads the user information.
+    const showModal = async (id: number): Promise<boolean> => {
+      applicationId = id;
+
+      // Call the modal method to show the modal.
+      return showModalInternal(applicationId);
```

Comment:
This method is just to show the modal here, no need to pass the `applicationId`.